beacon_node/eth2_libp2p/src/types/topics.rs

```diff
@@ -193,6 +193,12 @@ impl From<SubnetId> for GossipKind {
 
 // helper functions
 
+/// Get subnet id from an attestation subnet topic hash.
+pub fn subnet_id_from_topic_hash(topic_hash: &TopicHash) -> Option<SubnetId> {
```

Comment:
Rather than having two distinct sets of code to do this, I think we probably should use `GossipTopic::decode()` and then match on its Kind for the subnet. 

This way we don't have weird errors if one decoding logic is wrong but another is right. We localize the code.

Reply:
Yeah dunno why I didn't use `GossipTopic::decode()`. Will fix